aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -0,0 +1,295 @@
+                setVariable(authRule.getOwnerFieldOrDefault(), "String!", getUsername(authProvider));
+            }
+        }
+    }
+
+    private String getUsername(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
+        if (authProvider == null) {
+            throw new ApiException(
+                    "Attempted to subscribe to a model with owner based authorization without a Cognito provider",
+                    "Did you add the AWSCognitoAuthPlugin to Amplify before configuring it?"
+            );
+        }
+        String username = authProvider.getUsername();
+        if (username == null) {
+            throw new ApiException(
+                    "Attempted to subscribe to a model with owner based authorization without a username",
+                    "Make sure that a user is logged in before subscribing to a model with owner based auth"
+            );
+        }
+        return username;
+    }
+
+    private boolean isOwnerArgumentRequired(List<ModelOperation> operations) {
+        if (SubscriptionType.ON_CREATE.equals(operationType) && operations.contains(ModelOperation.CREATE)) {
+            return true;
+        }
+        if (SubscriptionType.ON_UPDATE.equals(operationType) && operations.contains(ModelOperation.UPDATE)) {
+            return true;
+        }
+        if (SubscriptionType.ON_DELETE.equals(operationType) && operations.contains(ModelOperation.DELETE)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     *  Returns String value used for GraphQL "query" in HTTP request body.
+     *
+     *  Sample return value:
+     *      subscription OnCreatePerson(owner: String!, nextToken: String) {
+     *            onCreatePerson(owner: $owner, nextToken: $nextToken) {
+     *               age dob first_name id last_name relationship owner
+     *            }
+     *       }
+     *
+     * @return String value used for GraphQL "query" in HTTP request body
+     */
+    @Override
+    public String getQuery() {
+        String inputTypeString = "";
+        String inputParameterString = "";
+        if (variableTypes.size() > 0) {
+            List<String> inputKeys = new ArrayList<>(variableTypes.keySet());
+            Collections.sort(inputKeys);
+
+            List<String> inputTypes = new ArrayList<>();
+            List<String> inputParameters = new ArrayList<>();
+            for (String key : inputKeys) {
+                inputTypes.add("$" + key + ": " + variableTypes.get(key));
+                inputParameters.add(key + ": $" + key);
+            }
+
+            inputTypeString = Wrap.inParentheses(TextUtils.join(", ", inputTypes));
+            inputParameterString = Wrap.inParentheses(TextUtils.join(", ", inputParameters));
+        }
+
+        String modelName = modelSchema.getName();
+        String operationString = new StringBuilder()
+                .append(Casing.from(Casing.CaseType.SCREAMING_SNAKE_CASE).to(Casing.CaseType.CAMEL_CASE)
```

Comment:
In hindsight, I was having too much fun when I made this `Casing` utility. But I think my intention was that you could import `CaseType` directly, for the most readability:
```java
Casing.from(CaseType.SCREAMING_SNAKE_CASE, ...)
```

Reply:
I'll go ahead and go a step further, and use static imports so I can just do this: `Casing.from(SCREAMING_SNAKE_CASE).to(PASCAL_CASE).convert(operationType.toString()))`

Reply:
Hmm, our checkstyle rules don't allow static imports, so I'm gonna leave this as is for now.